app/src/main/java/net/gsantner/opoc/frontend/filebrowser/GsFileBrowserFragment.java

```diff
@@ -236,7 +236,7 @@ private void updateMenuItems() {
         boolean selDirectoriesOnly = true;
         boolean selWritable = (!curFilepath.equals("/storage") && !curFilepath.equals("/storage/emulated"));
         boolean allSelectedFav = true;
-        final Collection<File> favFiles = _dopt.favouriteFiles != null ? _dopt.favouriteFiles : Collections.emptySet();
+        final Collection<File> favFiles = _appSettings.getFavouriteFiles();
```

Comment:
The _dopt is now assigned a function, would this not work here?

Reply:
Yeah, this is a little messy now. 

We want dopt and the adapter (and the file browser dialog) to be independent of markor so we have abstracted callbacks in the dopt. But the fragment needs more functionality and has to deal with appsettings directly.

I will look into restructuring this.